logs-proxy/proxy.yaml

```diff
@@ -0,0 +1,226 @@
+            )
+
+            return headers
+
+        def proxy_request(self, method: str):
+            """Proxy the request to S3 with authentication"""
+            if self.path == '/health':
+                self.send_response(200)
+                self.send_header('Content-Type', 'text/plain')
+                self.end_headers()
+                self.wfile.write(b'healthy\n')
+                return
+
+            # disallow directory listing
+            if self.path == '/':
+                self.send_response(403)
+                self.send_header('Content-Type', 'text/plain')
+                self.end_headers()
+                self.wfile.write(b'Forbidden: directory listing is not allowed\n')
+                return
+
+            s3_url = urllib.parse.urlparse(f'{S3_BUCKET_URL}{self.path}')
+            headers = self.sign_request(s3_url, method, {}, hashlib.sha256(b'').hexdigest())
+            request = urllib.request.Request(s3_url.geturl(), headers=headers, method=method)
+
+            try:
+                with urllib.request.urlopen(request) as response:
+                    # forward response
+                    self.send_response(response.getcode())
+                    # forward headers
+                    for header, value in response.headers.items():
+                        if header.lower() not in ['connection', 'transfer-encoding']:
+                            self.send_header(header, value)
+                    self.end_headers()
+                    # forward body
+                    self.wfile.write(response.read())
+
+            except urllib.error.HTTPError as e:
+                # forward HTTP errors
+                self.send_response(e.code)
+                for header, value in e.headers.items():
+                    if header.lower() not in ['connection', 'transfer-encoding']:
+                        self.send_header(header, value)
+                self.end_headers()
+                self.wfile.write(e.read())
+
+            except Exception as e:
+                # Internal server error
+                self.send_response(500)
+                self.send_header('Content-Type', 'text/plain')
+                self.end_headers()
+                self.wfile.write(b'Proxy error, see pod log\n')
+                sys.stderr.write(f'Error while proxying request: {str(e)}\n')
+
+        def do_GET(self):
+            self.proxy_request('GET')
+
+        def do_HEAD(self):
+            self.proxy_request('HEAD')
+
+        def log_request(self, code='-', size='-'):
+            """Log the request with X-Real-Ip and User-Agent headers"""
+            x_real_ip = self.headers.get('X-Real-Ip', '-')
+            user_agent = self.headers.get('User-Agent', '-')
+
+            sys.stderr.write(f'[{self.log_date_time_string()}] {x_real_ip} '
+                             f'"{self.requestline}" {code} {size} User-Agent="{user_agent}"\n')
+            sys.stderr.flush()
```

Comment:
I would maybe separate this into `stdout`? Logs aren't really errors

Reply:
Done.